test/form/samples/builtin-prototypes/check-tree-shake-on-regexp-if-it-is-used/_expected.js

```diff
@@ -0,0 +1,23 @@
+const normalRegexp_1 = /1/;
+const normalRegexp_2 = /1/;
+
+assert.strictEqual(normalRegexp_1.test('1'), true);
+assert.ok(normalRegexp_2.exec('1'));
```

Comment:
But in the `form` folder, these asserts do not have any effect because the code is not executed. So the problem here is that for every change, you still need a human to read the output and think hard if it is correct. If you put such a test in the `function` folder, it will execute the code with the asserts and you know it if is correct.
But there is nothing wrong with having tests both in the `function` and the `form` folder for the same feature.

Reply:
Yes, you are right, but in the `form` folder, I wrote these asserts just for using these RegRxps.

Reply:
I guess what I want to say is: If there is a real danger that Rollup breaks code e.g. by removing stuff it should not remove, one should always write a function test to ensure the output works in the specific case.
The advantage is that those tests not only document well why we want to keep certain code, they are also very stable and do not turn red if some unimportant aspect of the output changes like indentation etc.
`form` tests (and `chunking-form` tests) on the other hand suffer from all the issues that snapshots tests usually have i.e. they turn red very easily and then always require a human to manually review if the new output is still correct. And that is error-prone. There have been known instances where `form` tests for years asserted something wrong without anyone ever noticing.
On the other hand if you want to test that something is *removed*, then form tests are your only option and you need them. But there is nothing wrong with writing both, `function` tests to ensure important stuff is not removed and `form` tests to ensure unneeded stuff is removed.
But I guess in this concrete instance, it is not much of a difference and we can still merge it, but it is a general guiding principle to keep in the back of your head.

Reply:
Ok, I get it, thank you so much for your patience and guidance.